src/hooks/use-file-upload.tsx

```diff
@@ -153,12 +193,76 @@ export function useFileUpload({
 
   const resetBlocks = () => setContentBlocks([]);
 
+  /**
+   * Handle paste event for files (images, PDFs)
+   * Can be used as onPaste={handlePaste} on a textarea or input
+   */
+  const handlePaste = async (
+    e: React.ClipboardEvent<HTMLTextAreaElement | HTMLInputElement>,
+  ) => {
+    e.preventDefault();
+    const items = e.clipboardData.items;
+    if (!items) return;
+    const files: File[] = [];
+    for (let i = 0; i < items.length; i++) {
```

Comment:
need to turn on the eslint rule for this
```suggestion
    for (let i = 0; i < items.length; i += 1) {
```